net-mgmt/configsync/src/opnsense/mvc/app/models/VerbNetworks/ConfigSync/ConfigSync.xml

```diff
@@ -0,0 +1,68 @@
+<model>
+    <mount>//VerbNetworks/configsync</mount>
+    <description>Verb Networks "Configuration Sync" application</description>
+    <version>0.1.0</version>
+    <items>
+        <settings>
+
+            <Enabled type="BooleanField">
+                <default>0</default>
+                <Required>Y</Required>
+            </Enabled>
+
+            <Provider type="OptionField">
+                <Required>Y</Required>
+                <multiple>N</multiple>
+                <default>awss3</default>
+                <OptionValues>
+                    <awss3>Amazon Web Services S3</awss3>
+                </OptionValues>
+            </Provider>
+            
+            <ProviderKey type="TextField">
+                <Required>Y</Required>
+                <Mask>/^([a-zA-Z0-9]){16,128}$/u</Mask>
+                <ValidationMessage>Must provide a valid Storage Provider API key value</ValidationMessage>
+            </ProviderKey>
+            
+            <ProviderSecret type="TextField">
+                <Required>Y</Required>
+                <Mask>/^([a-zA-Z0-9\+\/]){16,128}$/u</Mask>
+                <ValidationMessage>Must provide a valid Storage Provider API key secret value</ValidationMessage>
+            </ProviderSecret>
+            
+            <StorageBucket type="TextField">
+                <default>[storage-bucket]</default>
+                <Required>Y</Required>
+                <Mask>/^[a-zA-Z0-9\-\.]{3,64}$/</Mask>
+                <ValidationMessage>Must provide a valid storage-bucket name</ValidationMessage>
+            </StorageBucket>
+            
+            <StoragePath type="TextField">
+                <default>[storage-path]</default>
+                <Required>Y</Required>
+                <Mask>/^[a-zA-Z0-9\!\-\_\.\*\'\@\:\$\=\+\;\,\/]{1,1024}$/u</Mask>
```

Comment:
I'd suggest that explicit escaping ensures the intended characters are included rather than the special meaning they may represent - this makes it easier to human-read and should not impact the speed/memory performance of the regex - I've attended the case insensitive match, removing the not-needed default but kept the escaping unless there is an easier way ...